migration/identity-migration/src/main/java/io/camunda/migration/identity/handler/sm/RoleMigrationHandler.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migration.identity.handler.sm;
+
+import static io.camunda.migration.identity.MigrationUtil.extractCombinedPermissions;
+import static io.camunda.migration.identity.config.sm.StaticEntities.getAuthorizationsByAudience;
+
+import io.camunda.migration.api.MigrationException;
+import io.camunda.migration.identity.client.ManagementIdentityClient;
+import io.camunda.migration.identity.dto.Role;
+import io.camunda.migration.identity.handler.MigrationHandler;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.RoleServices;
+import io.camunda.service.RoleServices.CreateRoleRequest;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RoleMigrationHandler extends MigrationHandler<Role> {
+
+  private final ManagementIdentityClient managementIdentityClient;
+  private final RoleServices roleServices;
+  private final AuthorizationServices authorizationServices;
+
+  private final AtomicInteger createdRoleCount = new AtomicInteger();
+  private final AtomicInteger totalRoleCount = new AtomicInteger();
+
+  public RoleMigrationHandler(
+      final Authentication authentication,
+      final ManagementIdentityClient managementIdentityClient,
+      final RoleServices roleServices,
+      final AuthorizationServices authorizationServices) {
+    this.managementIdentityClient = managementIdentityClient;
+    this.roleServices = roleServices.withAuthentication(authentication);
+    this.authorizationServices = authorizationServices;
+  }
+
+  @Override
+  protected List<Role> fetchBatch(final int page) {
+    // roles are not paginated in the Management Identity
```

Comment:
💭 This is actually more the rule than the exception, right? 😅

Reply:
Yes, my idea is to refactor this once we are done with all the different implementations 👍